x-pack/plugins/ml/public/embeddables/anomaly_swimlane/initialize_swim_lane_data_fetcher.ts

```diff
@@ -62,7 +63,14 @@ export const initializeSwimLaneDataFetcher = (
     swimlaneType: swimLaneApi.swimlaneType,
     viewBy: swimLaneApi.viewBy,
     perPage: swimLaneApi.perPage,
-    fromPage: swimLaneApi.fromPage,
+    fromPage: merge(
+      query$.pipe(map(() => 1)),
+      swimLaneApi.fromPage,
+      filters$.pipe(map(() => 1))
+    ).pipe(
+      distinctUntilChanged(),
+      tap((fromPage) => swimLaneApi.updatePagination({ fromPage }))
```

Comment:
I realized we should also reset the pagination on time filter changes. In the Anomaly Explorer page, we listen to filter, query and time range, and form a ES query. Then the pagination observable listens for the ES query change and reset pagination accordingly.  
And looking at this side effect I wonder if `fromPage` update should happen inside of `initializeSwimLaneControls` instead 🤔 In that case the swim lane fetcher logic only get the latest fromPage value, and the controls code is directly responsible for resetting it.

Reply:
Good point, updated in: [5a95ba0](https://github.com/elastic/kibana/pull/195090/commits/5a95ba0cd847f1cc290f05a9c89878d500d336bd)
I think it should be more convenient now